src/hooks/useBadgeContracts.ts

```diff
@@ -1,42 +1,38 @@
+import { DataKeys } from 'models/DataKeys'
 import { useMemo } from 'preact/hooks'
 import { useSnapshot } from 'valtio'
-import BaseBadgeContract from 'helpers/BaseBadgeContract'
-import ERC721BadgeContract from 'helpers/ERC721BadgeContract'
-import EmailBadgeContract from 'helpers/EmailBadgeContract'
-import Network from 'models/Network'
 import SealCredStore from 'stores/SealCredStore'
 
+export interface BaseBadgeContract {
+  type: DataKeys
+  derivative: string
+  original: string
+}
+
 export default function () {
-  const { emailLedger, ERC721Ledger, externalERC721Ledger } =
-    useSnapshot(SealCredStore)
+  const { ledgers } = useSnapshot(SealCredStore)
 
   return useMemo(() => {
-    let result: { [address: string]: BaseBadgeContract } = {}
-
-    result = Object.values(emailLedger).reduce(
-      (result, { derivativeContract, originalContract }) => ({
-        ...result,
-        [derivativeContract]: new EmailBadgeContract(
-          derivativeContract,
-          originalContract
-        ),
-      }),
-      result
-    )
-
-    return [
-      ...Object.values(ERC721Ledger),
-      ...Object.values(externalERC721Ledger),
-    ].reduce(
-      (result, { derivativeContract, originalContract }) => ({
-        ...result,
-        [derivativeContract]: new ERC721BadgeContract(
-          derivativeContract,
-          originalContract,
-          ERC721Ledger[originalContract] ? Network.Goerli : Network.Mainnet
+    return Object.entries(ledgers).reduce(
```

Comment:
Basically we're trying to fit to a type we've used before, maybe we can change `data`/`ledger`, so we don't need to change their types at all?